src/frontend/src/lib/utils/tokens.utils.ts

```diff
@@ -61,37 +57,15 @@ export const sortTokens = ({
  * In case of a tie, it sorts by token name and network name.
  *
  * @param $tokens - The list of tokens to sort.
- * @param $exchanges - The exchange rates for the tokens.
- * @param $balancesStore - The balances for the tokens.
  * @returns The sorted list of tokens.
  *
  */
-export const pinTokensWithBalanceAtTop = ({
-	$tokens,
-	$balancesStore
-}: {
-	$tokens: TokenUi[];
-	$balancesStore: CertifiedStoreData<BalancesData>;
-}): TokenUi[] => {
-	const tokensWithBalance: TokenWithBalance[] = $tokens.map(
-		(token) =>
-			({
-				...token,
-				balance: Number(
-					formatToken({
-						value: $balancesStore?.[token.id]?.data ?? BigNumber.from(0),
-						unitName: token.decimals,
-						displayDecimals: token.decimals
-					})
-				)
-			}) as TokenWithBalance
-	);
-
-	const [positiveBalances, nonPositiveBalances] = tokensWithBalance.reduce<
-		[TokenWithBalance[], TokenWithBalance[]]
-	>(
+export const pinTokensWithBalanceAtTop = ($tokens: TokenUi[]): TokenUi[] => {
+	const [positiveBalances, nonPositiveBalances] = $tokens.reduce<[TokenUi[], TokenUi[]]>(
 		(acc, token) => (
-			(token.usdBalance ?? 0) > 0 || token.balance > 0 ? acc[0].push(token) : acc[1].push(token),
+			(token.usdBalance ?? 0) > 0 || Number(token.formattedBalance ?? 0) > 0
```

Comment:
Why not use `balance`?

Reply:
no particular reason: before it was a number comparison, so I kept the same logic, but it is indifferent for me, no strong opinion

What would you prefer?

Reply:
I prefer avoiding casting as I'm always afraid to ends on `NaN` or other similar issue in some ways.

Reply:
perfect, changed!